packages/shared-component--linkList/src/linkList.html

```diff
@@ -1,32 +1,39 @@
-{% macro render(content, cms, config) -%}
-<div class="coop-c-list-container">
-  <h3 class="coop-c-list-container__title coop-t-h3">{{content.heading|safe}}</h3>
-  <ul class="coop-c-list-container__list">
-		{% for link in content.links %}
-			{% set linkDetails = {} %}
-			{% with p = cms.get_entry(link.sys.id) %}
-				{%- if p.type == 'internalLink' -%}
-					{% if p.fields.page.data %}
-						{% with internalLink = cms.get_entry(p.fields.page.data.sys.id) %}
-							{%- set _ = linkDetails.update({
-								"title" : p.fields.title.data,
-								"url"   : internalLink.full_url_path
-							})-%}
-						{%- endwith -%}
-					{%- endif -%}
-				{%- else -%}
-					{%- set _ = linkDetails.update({
-						"title" : p.fields.title.data,
-						"url"   : p.fields.url.data
-					})-%}
-				{%- endif -%}
-			<li>
-				<a class="coop-c-list-container__list-list-link" href="{% if p.type != 'externalLink' %}{{tenant_url_path}}{% endif %}{{linkDetails.url}}" data-contenttype="Link list" data-contentparent="{{content.heading}}" data-linktext="{{ linkDetails.title }}" >
-					{{linkDetails.title}}
-				</a>
-			</li>	
-			{% endwith %}
-		{% endfor %}
-  </ul>
-</div>
-{%- endmacro %}
\ No newline at end of file
+{%- import '/shared-component--link/src/link.html' as linkComponent -%}
+
+{%- macro render(content, cms, config) -%}
+
+	<div class="{{ 'coop-c-list-container coop-c-list-container--shadow' if content.dropShadow else 'coop-c-list-container'  }}">
+
+		{# By default, render heading here if not empty #}
+		{%- if content.heading and not headingListItem -%}
+			<h3 class="coop-c-list-container__title coop-t-h3">{{content.heading|safe}}</h3>
+		{%- endif -%}
+		
+		<ul class="coop-c-list-container__list">
+
+			{# If heading is set to render as non-clickable list item, load it here if not empty #}
+			{%- if content.heading and content.headingListItem -%}
+				<li class="coop-c-list-container__list-item">{{content.heading|safe}}</li>
+			{%- endif -%}
+
+			{%- for link in content.links -%}
+
+				{%- set linkItem = cms.get_entry(link.sys.id) -%}
+
+				{%- set config = {
+					'classNames': 'coop-c-list-container__list-list-link',
```

Comment:
I will have to make some changes to allow classNames config I think, there is probably a lot more of this elsewhere too isn't there